src/chains.rs

```diff
@@ -439,8 +435,39 @@ impl Rewrite for Chain {
         formatter.format_children(context, child_shape)?;
         formatter.format_last_child(context, shape, child_shape)?;
 
-        let result = formatter.join_rewrites(context, child_shape)?;
-        wrap_str(result, context.config.max_width(), shape)
+        formatter.join_rewrites(context, child_shape)
+    }
+}
+
+impl Rewrite for Chain {
+    fn rewrite(&self, context: &RewriteContext<'_>, shape: Shape) -> Option<String> {
+        debug!("rewrite chain {:?} {:?}", self, shape);
+
+        let result = self.format(context, shape)?;
```

Comment:
I do not think that this early return (`?`) is necessary. Worse, couldn't this prevent us from wrapping the chain with long elements?

Reply:
Yeah I think you are right. I'm still working on updating the `format` function to handle other scenarios (like when the last chain element exceeds the max_width) so I'll update this once I've finished out `format`